peer_serde.go

```diff
@@ -0,0 +1,45 @@
+// This file contains Protobuf and JSON serialization/deserialization methods for peer IDs.
+package peer
+
+import (
+	"encoding/json"
+
+	"github.com/golang/protobuf/proto"
```

Comment:
This pulls protobuf into quite a few deps. Given that it's *only* for a type assertion, I'd rather just leave it out for now.

Once we merge all the "core" deps into a single package, we can do stuff like this without complicating our lives quite as much.

---

If you think we should keep it, we'll have to add it to the `package.json`.

Reply:
👍 SGTM, will comment out the assertions to keep them as docs.